tests/dialects/test_linalg.py

```diff
@@ -36,6 +36,20 @@ def body(args: tuple[Any, ...]):
     func.FuncOp("foo", ([], []), funcBody)
 
 
+def test_matmul_on_memrefs():
+    @Builder.implicit_region(())
+    def funcBody(args: tuple[Any, ...]):
+        A = memref.Alloc.get(f32, shape=[100, 50])
+        B = memref.Alloc.get(f32, shape=[50, 100])
+        C = memref.Alloc.get(f32, shape=[100, 100])
+
+        linalg.MatmulOp(inputs=(A.results[0], B.results[0]), outputs=(C.results[0],))
+
+        func.Return()
+
+    func.FuncOp("foo", ([], []), funcBody)
```

Comment:
```suggestion
    A = memref.Alloc.get(f32, shape=[100, 50])
    B = memref.Alloc.get(f32, shape=[50, 100])
    C = memref.Alloc.get(f32, shape=[100, 100])

    matmul_op = linalg.MatmulOp(inputs=(A.results[0], B.results[0]), outputs=(C.results[0],))

    assert tuple(result.type for result in matmul_op.results) == ()
```
We still need to test that the result types are what we expect